Content.Server/Pinpointer/PinpointerSystem.cs

```diff
@@ -210,4 +275,80 @@ private Distance CalculateDistance(Vector2 vec, PinpointerComponent pinpointer)
         else
             return Distance.Far;
     }
+
+    /// <summary>
+    ///     Clears the list with stored targets and turns off the pinpointer.
+    /// </summary>
+    private void DeleteStoredTargets(EntityUid uid, PinpointerComponent component, EntityUid? user)
+    {
+        component.StoredTargets.Clear();
+        if (component.IsActive)
+        {
+            TogglePinpointer(uid, component);
+        }
+
+    }
+
+    /// <summary>
+    ///     Adds a verb that allows the user to search for the closest target containing a certain component, a verb that
+    ///     allows the user to select any of the stored targets and a verb that allows the user to clear the stored targets.
+    /// </summary>
+    private void OnPinpointerVerb(EntityUid uid, PinpointerComponent component, GetVerbsEvent<Verb> args)
+    {
+        if (!args.CanInteract || args.Hands == null)
+            return;
+
+        //Adds the closest target verb if there is at least 1 stored component.
+        if (component.Components.Count > 0)
+        {
+            foreach (var targetComponent in component.Components)
+            {
+                args.Verbs.Add(new Verb()
+                {
+                    Text = Loc.GetString(component.ComponentNames[component.Components.IndexOf(targetComponent)]),
+                    Act = () => LocateTarget(uid, component, args.User, targetComponent),
+                    Priority = 100,
+                    Category = VerbCategory.SearchClosest,
+                });
+            }
+        }
+
+        var storedOrder = 0;
+
+        //Adds the target selection verb if there is more than 1 stored target.
+        if (component.StoredTargets.Count > 1)
+        {
+            foreach (var target in component.StoredTargets)
+            {
+                if (Deleted(target))
+                {
+                    continue;
+                }
```

Comment:
This is going to cause serialization problems if targets are not being cleaned up properly.

Reply:
How would I go about preventing the possible serialization problems?